docs/product/roadmap.rst

```diff
@@ -1,6 +1,237 @@
+- Import data via any supported import method and have the types of data automatically detected during the import process
+
+  - Types are suggestions and can be changed by the user.
+
+Boolean Type
+------------
+- Add new type (use Postgres type if possible):
+
+  - Boolean
+- Autodetect this type during import
+- Allow user to change columns to this type
+- Add additional grouping options by value (yes/no)
+
+New Text Types
+--------------
+- Add new types:
+
+  - Email
+  - URL
+- Autodetect these types during import
+- Allow user to change columns to these types
+- Add additional grouping options:
+
+  - Email: Domain
+  - URL: TLD, Protocol
+
+Numeric Types
+-------------
+- Add new types, using existing Postgres types where possible:
+
+  - Money (with specified currency)
+  - Percentage
+  - Number
+- Autodetect these types during import
+- Allow user to change columns to these types
+- Add additional grouping options:
+
+  - Number: Range (calculate range options dynamically based on data, e.g. if data varies from 1-100, ranges could be 1-10, 10-20, etc.)
+
+  - Money: Range, Currency
+  - Percentage: Range
+
+Date and Time Types
+-------------------
+- Add new types, using existing Postgres types where possible:
+
+  - Date & Time
+  - Date
+  - Time
+  - Duration
+- Autodetect these types during import
+- Allow user to change columns to these types
+- Add additional grouping options:
+
+  - Date & Time, Date, Time support all grouping options supported by Postgres EXTRACT function.
+  - Duration: Range
+- Allow filtering using natural language for dates (e.g. "next month")
+
+Location Type
+-------------
+- Add new type, using existing PostGIS type where possible:
+
+  - Location
+- Autodetect this type during import
+- Allow user to change columns to this type
+- Add additional grouping options:
+
+  - Street Address
+  - Country
+  - Administrative Area Level 1 *(in the US, these are states)*
+  - Administrative Area Level 2 *(in the US, these are counties)*
+  - Administrative Area Level 3
+  - Administrative Area Level 4
+  - Administrative Area Level 5 
+  - Locality *(city/town)*
+  - Sublocality *(subdivision of city/town)*
+  - Neighborhood
+  - Postal Code
+  - Latitude
+  - Longitude
+
+The attributes of the location column type are based on results returned by the `Google Maps Geocoding API <https://developers.google.com/maps/documentation/geocoding/overview>`_, since they've done the work of putting addresses into a global format.
+
+Phone Number Type
+-----------------
+- Add new type
+ 
+  - Phone Number
+- Autodetect this type during import
+- Allow user to change columns to this type
+- Add grouping options:
+
+  - Country Code
+  - Area Code
+
+Relationship Type
+-----------------
+Users should be able to:
+
+- Create a column that represents a relationship to another record (e.g. Book --> Author)
+- "Extract" a column from a table into a separate table (change the underlying schemas)
```

Comment:
We can do sets of columns with what's already in the prototype.  It's no more difficult (at that layer) to do sets than single columns.  In fact, if I wanted an 'extract a single column' function done ASAP in the back end, it'd just be extract a set of one.